lib/analysis.js

```diff
@@ -52,6 +53,7 @@ function AnalysisFactory(DatabaseServiceClass) {
 
 AnalysisFactory.prototype.create = function(configuration, definition, callback) {
     debug('Using configuration=%j', configuration);
+    var logger = createLogger(configuration.logger).child({ analysis: definition.type, user: configuration.user });
```

Comment:
The problem with this approach is that it only covers the top level analysis. It will miss the rest of the analyses in the graph.

If my analysis is:

```js
{
  type: "data-observatory-measure",
  params: {
    source: {
      type: "buffer",
      params: {
         source: { type: "source" }
      }
    }
  }
}
```

It will only log the "data-observatory-measure" analysis, missing the "buffer" and "source" ones.